soroban-env-host/src/host/data_helper.rs

```diff
@@ -97,10 +117,16 @@ impl Host {
         contract_id: Hash,
         key: &Rc<LedgerKey>,
     ) -> Result<(), HostError> {
+        let body = ContractDataEntryBody::DataEntry(ContractDataEntryData {
+            val: ScVal::ContractExecutable(executable),
+            flags: 0,
+        });
         let data = LedgerEntryData::ContractData(ContractDataEntry {
             contract_id,
             key: ScVal::LedgerKeyContractExecutable,
-            val: ScVal::ContractExecutable(executable),
+            body,
+            type_: ContractDataType::Recreatable,
+            expiration_ledger_seq: self.with_ledger_info(|li| Ok(li.sequence_number))? + 4096, //TODO: use constant? Ideally we'd use the settings
```

Comment:
This TODO seems like something we definitely need to handle! Maybe thread the duration back to the caller? Also are we sure this should be Recreatable rather than unique?

Reply:
This should be unique. I also think we should just set `expiration_ledger_seq` to 0 and core can set the expiration ledger. What do you think?

Reply:
I see you did this in the update to the PR (set the expiration to zero) but .. I don't understand why. I mean, is it because callers of this path aren't in a position to specify the expiry time of a ContractExecutable LE?

Reply:
The host isn't aware of the min expiration amount. There's a bug here though - this code path can be called to update an existing executable, in which case the expiration ledger will re-written.

Reply:
this got fixed in followup today, thanks